postreise/plot/plot_generation_ts_stack.py

```diff
@@ -0,0 +1,280 @@
+                capacity_storage_ts, time_range[0], time_range[1]
+            )
+        if time_freq != "H":
+            pg_storage = resample_time_series(pg_storage, time_freq)
+            capacity_storage_ts = resample_time_series(capacity_storage_ts, time_freq)
+
+        pg_stack["storage"] = pg_storage.clip(lower=0)
+        capacity_ts += capacity_storage_ts
+
+        fig, (ax, ax_storage) = plt.subplots(
+            2,
+            1,
+            figsize=(20, 15),
+            sharex="row",
+            gridspec_kw={"height_ratios": [3, 1], "hspace": 0.02},
+        )
+        plt.subplots_adjust(wspace=0)
+        if normalize:
+            pg_storage = pg_storage.divide(capacity_storage_ts, axis="index")
+            ax_storage.set_ylabel("Normalized Storage", fontsize=label_fontsize)
+        else:
+            ax_storage.set_ylabel("Energy Storage (MW)", fontsize=label_fontsize)
+
+        ax_storage = pg_storage.plot(color=type2color["storage"], lw=4, ax=ax_storage)
+        ax_storage.fill_between(
+            pg_storage.index.values,
+            0,
+            pg_storage.values,
+            color=type2color["storage"],
+            alpha=0.5,
+        )
+
+        # Erase year in xticklabels
+        xt_with_year = list(ax_storage.__dict__["date_axis_info"][0])
+        xt_with_year[-1] = b"%b"
+        ax_storage.__dict__["date_axis_info"][0] = tuple(xt_with_year)
+
+        ax_storage.tick_params(axis="both", which="both", labelsize=tick_fontsize)
+        ax_storage.set_xlabel("")
+        for a in fig.get_axes():
+            a.label_outer()
+    else:
+        fig = plt.figure(figsize=(20, 10))
+        ax = fig.gca()
+
+    if normalize:
+        pg_stack = pg_stack.divide(capacity_ts, axis="index")
+        demand = demand.divide(capacity_ts, axis="index")
+        net_demand = net_demand.divide(capacity_ts, axis="index")
+        ax.set_ylabel("Normalized Generation", fontsize=label_fontsize)
+    else:
+        pg_stack = pg_stack.divide(1000000, axis="index")
+        demand = demand.divide(1000000, axis="index")
+        net_demand = net_demand.divide(1000000, axis="index")
+        ax.set_ylabel("Daily Energy TWh", fontsize=label_fontsize)
+
+    available_resources = [r for r in resources if r in pg_stack.columns]
+    pg_stack[available_resources].clip(0, None).plot.area(
+        color=type2color, linewidth=0, alpha=0.7, ax=ax, sharex="row"
+    )
+
+    if show_demand:
+        demand.plot(color="red", lw=4, ax=ax)
+    if show_net_demand:
+        net_demand.plot(color="red", ls="--", lw=2, ax=ax)
+
+    if not title:
+        title = area
+    ax.set_title("%s" % title, fontsize=title_fontsize)
+    ax.grid(color="black", axis="y")
+
+    if "storage" not in resources:
+        # Erase year in xticklabels
+        xt_with_year = list(ax.__dict__["date_axis_info"][0])
+        xt_with_year[-1] = b"%b"
+        ax.__dict__["date_axis_info"][0] = tuple(xt_with_year)
+        ax.set_xlabel("")
```

Comment:
Why do we only erase the year when storage is present?

Reply:
Actually, we did that in both situations, with or without storage. The reason for this code duplication is simply because with storage, the storage axis is the main x-axis, without storage, the original axis is the main x-axis. We only show xticklabels on the main axis.